qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -159,7 +159,7 @@ def _layer_permutation(self, layer_partition, layout, qubit_subset, coupling, tr
         logger.debug("layer_permutation: gates = %s", gates)
 
         # Can we already apply the gates? If so, there is no work to do.
-        dist = sum(coupling.distance(layout[g[0]], layout[g[1]]) for g in gates)
+        dist = sum(coupling._dist_matrix[layout._v2p[g[0]], layout._v2p[g[1]]] for g in gates)
```

Comment:
Same comments as for sabre_swap.

Reply:
I think here it would actually be a bit slower. Since `self.dist_matrix` would be a non-slotted instance variable getattr vs a slotted access of a local on each iteration. The other places I think this makes sense since they're all accessing an instance variable, this is the only one which is different because stochastic swap is passing around the coupling map instead of using an instance variable.

Reply:
I added a comment about that here: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/8046f43ce5a63a0ccd6035e45206bafd9ec6a640